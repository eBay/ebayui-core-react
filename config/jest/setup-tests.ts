import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
