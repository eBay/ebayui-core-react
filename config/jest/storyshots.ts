import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
