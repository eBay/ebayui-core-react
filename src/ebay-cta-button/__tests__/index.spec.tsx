import React from 'react'
import requireContext from 'node-require-context'

import { render } from '@testing-library/react';
import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayCtaButton } from '../index'

initStoryshots({
    config: ({ configure }) => {
        const req = requireContext('./', false, /\.stories\.tsx$/);
        return configure(req, module)
    }

})

describe('<EbayCtaButton>', () => {
    describe('on passing a ref', () => {
        it('should set the ref to a element', () => {
            const ref = React.createRef() as any
            render(<EbayCtaButton href="https://www.ebay.com" ref={ref} />)

            expect(ref.current.tagName).toBe('A')
        })
    })
})
