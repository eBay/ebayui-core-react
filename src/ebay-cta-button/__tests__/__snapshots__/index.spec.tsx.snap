// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-cta-button Default 1`] = `
<DocumentFragment>
  <p>
    <a
      class="cta-btn"
      href="https://ebay.com"
    >
      <span
        class="cta-btn__cell"
      >
        <span>
          Take Action Now!
        </span>
        <svg
          aria-hidden="true"
          class="icon icon--cta"
          focusable="false"
          height="8"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-cta"
          />
        </svg>
      </span>
    </a>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-cta-button Fluid 1`] = `
<DocumentFragment>
  <p>
    <a
      class="cta-btn cta-btn--fluid"
      href="https://ebay.com"
    >
      <span
        class="cta-btn__cell"
      >
        <span>
          100%
        </span>
        <svg
          aria-hidden="true"
          class="icon icon--cta"
          focusable="false"
          height="8"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-cta"
          />
        </svg>
      </span>
    </a>
  </p>
  <p>
    <a
      class="cta-btn cta-btn--large cta-btn--fluid"
      href="https://ebay.com"
    >
      <span
        class="cta-btn__cell"
      >
        <span>
          Large!
        </span>
        <svg
          aria-hidden="true"
          class="icon icon--cta"
          focusable="false"
          height="8"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-cta"
          />
        </svg>
      </span>
    </a>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-cta-button Large 1`] = `
<DocumentFragment>
  <p>
    <a
      class="cta-btn cta-btn--large"
      href="https://ebay.com"
    >
      <span
        class="cta-btn__cell"
      >
        <span>
          Large Button
        </span>
        <svg
          aria-hidden="true"
          class="icon icon--cta"
          focusable="false"
          height="8"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-cta"
          />
        </svg>
      </span>
    </a>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-cta-button Truncated 1`] = `
<DocumentFragment>
  <div
    style="max-width: 200px;"
  >
    <p>
      <a
        class="cta-btn cta-btn--truncated"
        href="https://ebay.com"
      >
        <span
          class="cta-btn__cell"
        >
          <span>
            Wide Long Call To Action!
          </span>
          <svg
            aria-hidden="true"
            class="icon icon--cta"
            focusable="false"
            height="8"
            width="8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-cta"
            />
          </svg>
        </span>
      </a>
    </p>
    <p>
      <a
        class="cta-btn cta-btn--large cta-btn--truncated"
        href="https://ebay.com"
      >
        <span
          class="cta-btn__cell"
        >
          <span>
            Go Big with Call To Action!
          </span>
          <svg
            aria-hidden="true"
            class="icon icon--cta"
            focusable="false"
            height="8"
            width="8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-cta"
            />
          </svg>
        </span>
      </a>
    </p>
  </div>
</DocumentFragment>
`;
